src/rpc/handlers/VaultInfo.hpp

```diff
@@ -0,0 +1,134 @@
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include "data/BackendInterface.hpp"
+#include "rpc/Errors.hpp"
+#include "rpc/JS.hpp"
+#include "rpc/common/MetaProcessors.hpp"
+#include "rpc/common/Specs.hpp"
+#include "rpc/common/Types.hpp"
+#include "rpc/common/Validators.hpp"
+
+#include <boost/json/conversion.hpp>
+#include <boost/json/object.hpp>
+#include <boost/json/value.hpp>
+#include <xrpl/protocol/STLedgerEntry.h>
+#include <xrpl/protocol/jss.h>
+
+#include <cstdint>
+#include <memory>
+#include <optional>
+#include <string>
+
+namespace rpc {
+
+/**
+ * @brief The vault_info command retrieves information about a vault, currency, shares etc.
+ */
+class VaultInfoHandler {
+    std::shared_ptr<BackendInterface> sharedPtrBackend_;
+
+public:
+    /**
+     * @brief Construct a new VaultInfo object
+     *
+     * @param sharedPtrBackend The backend to use
+     */
+    VaultInfoHandler(std::shared_ptr<BackendInterface> const& sharedPtrBackend);
+
+    /**
+     * @brief A struct to hold the input data for the command
+     */
+    struct Input {
+        std::optional<std::string> vaultID;
+        std::optional<std::string> owner;
+        std::optional<uint32_t> tnxSequence;
+        std::optional<uint32_t> ledgerIndex;
+    };
+
+    /**
+     * @brief A struct to hold the output data for the command
+     */
+    struct Output {
+        boost::json::value vault;
+        uint32_t ledgerIndex{};
+        bool validated = true;
+    };
+
+    using Result = HandlerReturnType<Output>;
+
+    /**
+     * @brief Returns the API specification for the command
+     *
+     * @param apiVersion The api version to return the spec for
+     * @return The spec for the given apiVersion
+     */
+    static RpcSpecConstRef
+    spec([[maybe_unused]] uint32_t apiVersion)
+    {
+        static auto const kRPC_SPEC = RpcSpec{
+            {JS(vault_id),
+             meta::WithCustomError{
+                 validation::CustomValidators::uint256HexStringValidator, Status(ClioError::RpcMalformedRequest)
+             }},
+            {JS(owner),
+             meta::WithCustomError{
+                 validation::CustomValidators::accountBase58Validator,
+                 Status(ClioError::RpcMalformedRequest, "OwnerNotHexString")
+             }},
+            {JS(seq), meta::WithCustomError{validation::Type<uint32_t>{}, Status(ClioError::RpcMalformedRequest)}},
+            {JS(ledger_index), validation::CustomValidators::ledgerIndexValidator},
+        };
+
+        return kRPC_SPEC;
+    }
+
+    /**
+     * @brief Process the VaultInfo command
+     *
+     * @param input The input data for the command
+     * @param ctx The context of the request
+     * @return The result of the operation
+     */
+    Result
+    process(Input input, Context const& ctx) const;
```

Comment:
```suggestion
    process(Input const& input, Context const& ctx) const;
```

I think it's worth changing everywhere where applicable

Reply:
I think this should be changed in a separate PR for all handlers or in `DefaultProcessor`.
I created an issue for that: https://github.com/XRPLF/clio/issues/2265

Reply:
Sure